mitiq/shadows/shadows_utils.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Defines utility functions for classical shadows protocol."""
+from typing import Tuple, List, Any
+from numpy.typing import NDArray
+import cirq
+import numpy as np
+from cirq.ops.pauli_string import PauliString
+
+
+def min_n_total_measurements(epsilon: float, num_qubits: int) -> int:
```

Comment:
Ah, this `min_n_total_measurements` refers to the minimum number of classical shadows required for state reconstruction. It only takes one variable epsilon, the error rate of state reconstruction. The `calculated_shadow_bound` function takes two variables, error rate, and accuracy, and it calculates both the minimum number of classical shadows required and the number of groups "K" into which we need to split the shadows for achieving the desired accuracy in operator expectation value estimation.

Reply:
Thanks for explaining this. Could you also add this explanation in the docstrings of the two functions?
For example you could say:

- _This function returns the minimum number of classical shadows required for state reconstruction._

- _This function returns the minimum number of classical shadows required and the number of groups "K" into which we need to split the shadows for achieving the desired accuracy in operator expectation value estimation._

Optionally, you could rename the two functions accordingly:

- `min_n_total_measurements` -> `sampling_cost_for_tomography`
- `calculate_shadow_bound` -> `sampling_cost_for_estimation`

Reply:
Thanks for your suggestion, Andrea!  I have applied your suggestion to the doc string!  However, I'm not sure whether the name should be changed to `sampling_cost_for_*` since the sampling cost seems not clear for what it was used for, both of them are a lower bound on measurements/snapshots for a particular target... I'll come up with some other name if this name looks confusing to you.

Reply:
That would be helpful. Sorry for me complaining often about names, it seems a silly aspect but good names are very  important for making open source code easy to understand and to maintain.

Reply:
I see, thanks for explain Andrea! How about 
get_shadow_bound_tomography
get_shadow_bound_operator

or

n_measurements_tomography_bound
n_measurements_opratorexpectation_bound

Since both of them are in fact bounds of number of measurements for achieving the aimed error rate and failure rate.

Reply:
[Naming things is notoriously hard](https://www.martinfowler.com/bliki/TwoHardThings.html), but as Andrea said, extremely important!

---

IMO if an int is being returned, having `bound` at the **end** of the function name makes much more sense than `tomography`/`operator` as I would expect those to return different things.

Reply:
Thanks!  I will try my best to come up with names that work for all of us!